### PR TITLE
Deprecated support for delaying state recovery pending master nodes

### PR DESCRIPTION
**Pull request by ShawnLi1014**
As mentioned in issue [#51806](https://github.com/elastic/elasticsearch/issues/51806), we now have the option to delay state recovery until a certain number of master-eligible nodes have joined, but this seems unnecessary: we require a majority of master-eligible nodes for state recovery, but there is no advantage in waiting for more.
This pull request deprecated the following settings in `GatewayService.java`
* `gateway.expected_nodes`
* `gateway.expected_master_nodes`
* `gateway.recover_after_nodes`
*  `gateway.recover_after_master_nodes`

### CODE REVIEW

File: docs/reference/modules/gateway.asciidoc [hunk 1]
```diff
@@ -13,12 +13,14 @@ recover the cluster state and the cluster's data:
     The number of (data or master) nodes that are expected to be in the cluster.
```
Comment (DaveCTurner) [inline docs/reference/modules/gateway.asciidoc]:
```suggestion
    deprecated[7.7,This setting will be removed in 8.0. You should use `gateway.expected_data_nodes` instead.]
    The number of (data or master) nodes that are expected to be in the cluster.
```

File: docs/reference/modules/gateway.asciidoc [hunk 2]
```diff
@@ -13,12 +13,14 @@ recover the cluster state and the cluster's data:
...
     Recovery of local shards will start as soon as the expected number of
     nodes have joined the cluster. Defaults to `0`
+    Deprecated and will be removed in 8.0
 
 `gateway.expected_master_nodes`::
 
     The number of master nodes that are expected to be in the cluster.
```
Comment (DaveCTurner) [inline docs/reference/modules/gateway.asciidoc]:
```suggestion
    deprecated[7.7,This setting will be removed in 8.0. You should use `gateway.expected_data_nodes` instead.]
    The number of master nodes that are expected to be in the cluster.
```

File: docs/reference/modules/gateway.asciidoc [hunk 3]
```diff
@@ -38,10 +40,12 @@ as long as the following conditions are met:
 `gateway.recover_after_nodes`::
 
     Recover as long as this many data or master nodes have joined the cluster.
```
Comment (DaveCTurner) [inline docs/reference/modules/gateway.asciidoc]:
```suggestion
    deprecated[7.7,This setting will be removed in 8.0. You should use `gateway.recover_after_data_nodes` instead.]
    Recover as long as this many data or master nodes have joined the cluster.
```

File: docs/reference/modules/gateway.asciidoc [hunk 4]
```diff
@@ -38,10 +40,12 @@ as long as the following conditions are met:
...
+    Deprecated and will be removed in 8.0
 
 `gateway.recover_after_master_nodes`::
 
     Recover as long as this many master nodes have joined the cluster.
```
Comment (DaveCTurner) [inline docs/reference/modules/gateway.asciidoc]:
```suggestion
    deprecated[7.7,This setting will be removed in 8.0. You should use `gateway.recover_after_data_nodes` instead.]
    Recover as long as this many master nodes have joined the cluster.
```

File: docs/reference/modules/gateway.asciidoc [hunk 5]
```diff
@@ -13,12 +13,14 @@ recover the cluster state and the cluster's data:
     The number of (data or master) nodes that are expected to be in the cluster.
     Recovery of local shards will start as soon as the expected number of
     nodes have joined the cluster. Defaults to `0`
+    Deprecated and will be removed in 8.0
```
Comment (DaveCTurner) [inline docs/reference/modules/gateway.asciidoc]:
```suggestion
```

File: docs/reference/modules/gateway.asciidoc [hunk 6]
```diff
@@ -13,12 +13,14 @@ recover the cluster state and the cluster's data:
...
 
 `gateway.expected_master_nodes`::
 
     The number of master nodes that are expected to be in the cluster.
     Recovery of local shards will start as soon as the expected number of
     master nodes have joined the cluster. Defaults to `0`
+    Deprecated and will be removed in 8.0
```
Comment (DaveCTurner) [inline docs/reference/modules/gateway.asciidoc]:
```suggestion
```

File: docs/reference/modules/gateway.asciidoc [hunk 7]
```diff
@@ -38,10 +40,12 @@ as long as the following conditions are met:
 `gateway.recover_after_nodes`::
 
     Recover as long as this many data or master nodes have joined the cluster.
+    Deprecated and will be removed in 8.0
```
Comment (DaveCTurner) [inline docs/reference/modules/gateway.asciidoc]:
```suggestion
```

File: docs/reference/modules/gateway.asciidoc [hunk 8]
```diff
@@ -38,10 +40,12 @@ as long as the following conditions are met:
...
 
 `gateway.recover_after_master_nodes`::
 
     Recover as long as this many master nodes have joined the cluster.
+    Deprecated and will be removed in 8.0
```
Comment (DaveCTurner) [inline docs/reference/modules/gateway.asciidoc]:
```suggestion
```

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
I suggested an alternative way to note that these are deprecated in the docs, but the code changes look good.

**Comment by DaveCTurner** [comment]
@elasticmachine test this please

**Comment by DaveCTurner** [comment]
@elasticmachine update branch

**Comment by DaveCTurner** [comment]
@elasticmachine test this please

**Comment by DaveCTurner** [comment]
@elasticmachine test this please

**Review by DaveCTurner — APPROVED** [review]
LGTM thanks @ShawnLi1014. Do you plan to open a follow-up to remove these settings?

**Comment by ShawnLi1014** [comment]
> LGTM thanks @ShawnLi1014. Do you plan to open a follow-up to remove these settings?

Thank you, and yes I am working on it

**Pull request merged**